apps/internal/client/client.go

```diff
@@ -0,0 +1,309 @@
+	Credential  *accesstokens.Credential
+}
+
+// AuthenticationResult contains the results of one token acquisition operation in PublicClientApplication
+// or ConfidentialClientApplication. For details see https://aka.ms/msal-net-authenticationresult
+type AuthenticationResult struct {
+	Account        shared.Account
+	IDToken        accesstokens.IDToken
+	AccessToken    string
+	ExpiresOn      time.Time
+	GrantedScopes  []string
+	DeclinedScopes []string
+}
+
+// CreateAuthenticationResultFromStorageTokenResponse creates an authenication result from a storage token response (which is generated from the cache).
+func CreateAuthenticationResultFromStorageTokenResponse(storageTokenResponse storage.StorageTokenResponse) (AuthenticationResult, error) {
+	if err := storageTokenResponse.AccessToken.Validate(); err != nil {
+		return AuthenticationResult{}, fmt.Errorf("problem with access token in StorageTokenResponse: %w", err)
+	}
+
+	account := storageTokenResponse.Account
+	accessToken := storageTokenResponse.AccessToken.Secret
+	expiresOn, err := convertStrUnixToUTCTime(storageTokenResponse.AccessToken.ExpiresOnUnixTimestamp)
+	if err != nil {
+		return AuthenticationResult{}, fmt.Errorf("token response from server is invalid because expires_in is set to %q", storageTokenResponse.AccessToken.ExpiresOnUnixTimestamp)
+	}
+	grantedScopes := strings.Split(storageTokenResponse.AccessToken.Scopes, scopeSeparator)
+
+	// Checking if there was an ID token in the cache; this will throw an error in the case of confidential client applications.
+	var idToken accesstokens.IDToken
+	if !storageTokenResponse.IDToken.IsZero() {
+		idToken, err = accesstokens.NewIDToken(storageTokenResponse.IDToken.Secret)
+		if err != nil {
+			return AuthenticationResult{}, err
+		}
+	}
+	return AuthenticationResult{account, idToken, accessToken, expiresOn, grantedScopes, nil}, nil
+}
+
+// CreateAuthenticationResult creates an AuthenticationResult.
+// TODO(jdoak): (maybe, we did a refactor): Make this a method on TokenResponse() that takes only 1 arge, Account.
+func CreateAuthenticationResult(tokenResponse accesstokens.TokenResponse, account shared.Account) (AuthenticationResult, error) {
+	if len(tokenResponse.DeclinedScopes) > 0 {
+		return AuthenticationResult{}, fmt.Errorf("token response failed because declined scopes are present: %s", strings.Join(tokenResponse.DeclinedScopes, ","))
+	}
+	return AuthenticationResult{
+		Account:       account,
+		IDToken:       tokenResponse.IDToken,
+		AccessToken:   tokenResponse.AccessToken,
+		ExpiresOn:     tokenResponse.ExpiresOn,
+		GrantedScopes: tokenResponse.GrantedScopes,
+	}, nil
+}
+
+//GetAccessToken returns the access token of the authentication result
+func (ar AuthenticationResult) GetAccessToken() string {
+	return ar.AccessToken
+}
+
+// GetAccount returns the account of the authentication result
+func (ar AuthenticationResult) GetAccount() shared.Account {
+	return ar.Account
+}
```

Comment:
Are these methods necessary?

Reply:
Likely not, these two fields should be fine to get off of the AuthenticationResult

Reply:
Removed

Reply:
They are removed.